pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -359,6 +357,12 @@ public boolean isPartialUpsertEnabled() {
         && _tableUpsertMetadataManager.getUpsertMode() == UpsertConfig.Mode.PARTIAL;
   }
 
+  private boolean isUpsertPreloadEnabled() {
+    UpsertConfig upsertConfig = _tableConfig.getUpsertConfig();
+    return _tableUpsertMetadataManager != null && _segmentPreloadExecutor != null && upsertConfig != null
```

Comment:
Not introduced in this PR, but do we have checks in other places? Basically we don't want it to skip preloading silently

Reply:
Good question. I followed the method call below used to create partition mgr, as that's the entry point for preloading. And this is only called in `addSegment()` or `handleUpsert()` methods. So afaik, we should be good to check this at the beginning of the addSegment() method.
```
... _tableUpsertMetadataManager.getOrCreatePartitionManager(partitionId);
```